### PR TITLE
internal/charmstore: return blob hash when downloading a charm archive

### PR DESCRIPTION
**Pull request by rogpeppe**
We want clients to get the hash as well as the data.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/395/

File: params/params.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ import (
 	"gopkg.in/juju/charm.v4"
 )
 
+// ContentHashHeader specifies the header attribute
+// that will hold the content hash for archive GET responses.
+const ContentHashHeader = "x-content-sha384"
```
Comment (frankban) [inline params/params.go:17]: After reading http://stackoverflow.com/questions/3561381/custom-http-headers-naming-conventions
I'd suggest to rename this to something like "Content-SHA384". [shrug]

**Comment by frankban** [comment]
Hey Roger, thanks for this.
:+1:

**Comment by jrwren** [comment]
:+1:

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**